src/catchup/ApplyBucketsWork.cpp

```diff
@@ -119,6 +119,13 @@ ApplyBucketsWork::onReset()
             addBucket(getBucket(hsb.snap));
             addBucket(getBucket(hsb.curr));
         }
+        // estimate the number of ledger entries contained in those buckets
+        // use accounts as a rought approximator as to overestimate a bit
+        size_t constexpr estimatedLedgerEntrySize = sizeof(AccountEntry);
```

Comment:
This is going to be a very bad estimate because it won't account for the dynamic size of `AccountEntry`. If you want something a bit more reasonable, you might try constructing a typical `AccountEntry ae` and use `xdr_traits<AccountEntry>::serial_size(ae)`.

Reply:
This estimate (using a default constructed `AccountEntry`) is "good enough" for this purpose:
* we want to over-estimate the number of ledger entries, so under-estimating the size of a `BucketEntry` as the size of a default constructed `BucketEntry` is good enough as a default constructed `AccountEntry` already contains things like `AccountID` (fixed array).
* I agree that using `xdr_traits<>::serial_size()` is better as it doesn't make any assumption on how xvectors are stored in memory for example.